### PR TITLE
PushOver notifications

### PR DESCRIPTION
**Pull request by tdk1069**
quick copy of an existing notifications scripts modified to work with PushOver notification app. I have been running this a while at level 3 to test. But it is also my first one or acme so feel free to rip apart!

### CODE REVIEW

File: notify/pushover.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
```
Comment (Neilpang) [inline notify/pushover.sh]: fix the shebang same at others

File: notify/pushover.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  PUSHOVER_USER="${PUSHOVER_USER:-$(_readaccountconf_mutable PUSHOVER_USER)}"
+  if [ -z "$PUSHOVER_USER" ]; then
+    PUSHOVER_USER=""
+    _err "You didn't specify a PushOver UserKey yet."
+    return 1
+  fi
+  _saveaccountconf_mutable PUSHOVER_USER "$PUSHOVER_USER"
+
+  PUSHOVER_SOUND="${PUSHOVER_SOUND:-$(_readaccountconf_mutable PUSHOVER_SOUND)}"
```
Comment (Neilpang) [inline notify/pushover.sh]: PUSHOVER_SOUND is overritten.
PUSHOVER_SOUND="intermission" 	# https://pushover.net/api#sounds

File: notify/pushover.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  if [ -z "$PUSHOVER_SOUND" ]; then
+    PUSHOVER_SOUND="" # Play default if not specified.
+  fi
+  _saveaccountconf_mutable PUSHOVER_SOUND "$PUSHOVER_SOUND"
+
+  _H1="Content-Type: application/json"
+
+  _data="{\"token\": \"$PUSHOVER_TOKEN\",\"user\": \"$PUSHOVER_USER\",\"message\": \"$_content\",\"sound\": \"$PUSHOVER_SOUND\"}"
```
Comment (Neilpang) [inline notify/pushover.sh]: only the `_content` is sent,  please add the `_subject` too.

**Comment by Neilpang** [comment]
please also fix he CI format errors.

**Comment by Neilpang** [comment]
please add the usage here: https://github.com/Neilpang/acme.sh/wiki/notify

And create a new issue for bug reporting, example:  https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide#10-please-create-a-new-issue-for-future-bugs

File: notify/pushover.sh (hunk 1 above)
Comment (tdk1069) [reply notify/pushover.sh]: updated to match the others, No idea where I got the other one from

File: notify/pushover.sh (hunk 2 above)
Comment (tdk1069) [reply notify/pushover.sh]: removed these as I was testing. Set via environment

File: notify/pushover.sh (hunk 3 above)
Comment (tdk1069) [reply notify/pushover.sh]: Didn't realise was a subject! Happily added to the pushover title field

File: notify/pushover.sh (hunk 1 above)
Comment (Neilpang) [reply notify/pushover.sh]: why not read another hook first ?

https://github.com/Neilpang/acme.sh/blob/master/notify/mail.sh#L1

File: notify/pushover.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env sh
+
+#Support PUSHOVER.com api
```
Comment (Neilpang) [inline notify/pushover.sh]: PUSHOVER.net  ?

File: notify/pushover.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  fi
+  _saveaccountconf_mutable PUSHOVER_USER "$PUSHOVER_USER"
+
+  PUSHOVER_SOUND="${PUSHOVER_SOUND:-$(_readaccountconf_mutable PUSHOVER_SOUND)}"
+  if [ -z "$PUSHOVER_SOUND" ]; then
+    PUSHOVER_SOUND="" # Play default if not specified.
+  fi
+  _saveaccountconf_mutable PUSHOVER_SOUND "$PUSHOVER_SOUND"
+
+  _data="{\"token\": \"$PUSHOVER_TOKEN\",\"user\": \"$PUSHOVER_USER\",\"title\": \"$_subject\",\"message\": \"$_content\",\"sound\": \"$PUSHOVER_SOUND\"}"
```
Comment (Neilpang) [inline notify/pushover.sh]: you should not use raw `_subject` and `_content` here.
they must be json-encoded.
https://github.com/Neilpang/acme.sh/blob/master/notify/slack.sh#L35

File: notify/pushover.sh (hunk 4 above)
Comment (tdk1069) [reply notify/pushover.sh]: updated... sorry, comment was from an existing file and I didn't double check the comments.

**Comment by Neilpang** [comment]
please add the usage here: https://github.com/Neilpang/acme.sh/wiki/notify

And create a new issue for bug reporting, example: https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide#10-please-create-a-new-issue-for-future-bugs

**Comment by Neilpang** [comment]
Can you please add an optional parameter ?
```
PUSHOVER_DEVICE     optional,  specifies the device to push to.  default is empty, push to all devices.

PUSHOVER_PRIORITY  optional,  speicfies the priority. 

```

see https://pushover.net/api

**Comment by i4to** [comment]
please fix the ci errors.

and update the usage: https://github.com/Neilpang/acme.sh/wiki/notify

**Comment by Neilpang** [comment]
please fix the ci errors.

and update the usage: https://github.com/Neilpang/acme.sh/wiki/notify

**Comment by Neilpang** [comment]
fix the ci errors.

**Comment by lcdtyph** [comment]
Could you please delete [this](https://github.com/tdk1069/acme.sh/blob/ea27fa69f2646c7a3f352088992ebc4df6436c98/notify/pushover.sh#L45) line to fix the CI issue?

I really need this feature, really appreciate it. Thank you.

**Comment by tdk1069** [comment]
Line removed. Sorry it’s taken so long, I wish I realised it was that easy!

**Pull request merged**